File: AuctionMarket/src/main/java/com/portfolio/auctionmarket/auth/controller/AuthController.java
package com.portfolio.auctionmarket.auth.controller;

import com.portfolio.auctionmarket.auth.dto.LoginRequest;
import com.portfolio.auctionmarket.auth.dto.RefreshTokenRequest;
import com.portfolio.auctionmarket.auth.dto.SecurityUser;
import com.portfolio.auctionmarket.auth.dto.TokenResponse;
import com.portfolio.auctionmarket.auth.service.AuthService;
import com.portfolio.auctionmarket.global.error.LoginException;
import com.portfolio.auctionmarket.global.response.ApiResponse;
import jakarta.servlet.http.HttpServletResponse;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.http.ResponseCookie;
import org.springframework.http.ResponseEntity;
import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.web.bind.annotation.*;

@RestController
@Slf4j
@RequiredArgsConstructor
@RequestMapping("/api/auth")
public class AuthController {

    private final AuthService authService;

    @PostMapping("/login")
    public ResponseEntity<ApiResponse<String>> login(@Valid @RequestBody LoginRequest request, HttpServletResponse response) {
        TokenResponse tokenResponse = authService.login(request);
        ResponseCookie responseCookie = ResponseCookie.from("refreshToken", tokenResponse.getRefreshToken())
                .path("/")
                .secure(false)
                .httpOnly(true)
                .sameSite("Lax")
                .maxAge(7 * 24 * 60 * 60)
                .build();

        response.setHeader("Set-Cookie", responseCookie.toString());
        return ResponseEntity.ok(ApiResponse.success("로그인 성공", tokenResponse.getAccessToken()));
    }

    @PostMapping("/logout")
    public ResponseEntity<ApiResponse<String>> logout(@AuthenticationPrincipal SecurityUser user) {
        authService.logout(user.getUserId());
        return ResponseEntity.ok(ApiResponse.success("로그아웃 성공"));
    }

    @PostMapping("/refresh")
    public ApiResponse<String> refresh(@CookieValue(value = "refreshToken", required = false) String refreshToken, HttpServletResponse response) {

        if (refreshToken == null) {
            throw new RuntimeException("리프레시 토큰이 없습니다.");
        }

        TokenResponse tokenResponse = authService.refreshAccessToken(refreshToken);
        ResponseCookie responseCookie = ResponseCookie.from("refreshToken", tokenResponse.getRefreshToken())
                .path("/")
                .secure(false)                                // https 환경에서만 쿠키가 발동합니다.
                .sameSite("Lax")                           // 동일 사이트과 크로스 사이트에 모두 쿠키 전송이 가능합니다
                .httpOnly(true)                              // 브라우저에서 쿠키에 접근할 수 없도록 제한
                .maxAge(7 * 24 * 60 * 60)
                .build();

        response.setHeader("Set-Cookie", responseCookie.toString());
        return ApiResponse.success("토큰 갱신 성공", tokenResponse.getAccessToken());
    }

}

File: AuctionMarket/auction-app-user/src/main/kotlin/org/example/auctioncommon/domain/auth/controller/AuthController.kt
package org.example.auctioncommon.domain.auth.controller

import jakarta.servlet.http.HttpServletResponse
import jakarta.validation.Valid
import org.example.auctioncommon.domain.auth.dto.LoginRequest
import org.example.auctioncommon.domain.auth.dto.SecurityUser
import org.example.auctioncommon.domain.auth.dto.TokenResponse
import org.example.auctioncommon.domain.auth.service.AuthService
import org.example.auctioncommon.global.response.ApiResponse
import org.springframework.http.ResponseCookie
import org.springframework.http.ResponseEntity
import org.springframework.security.core.annotation.AuthenticationPrincipal
import org.springframework.web.bind.annotation.CookieValue
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/auth")
class AuthController(
    private val authService: AuthService
) {


    @PostMapping("/login")
    fun login(
        @RequestBody request: @Valid LoginRequest,
        response: HttpServletResponse
    ): ResponseEntity<ApiResponse<String>> {
        val tokenResponse: TokenResponse = authService.login(request)
        val responseCookie: ResponseCookie = ResponseCookie.from("refreshToken", tokenResponse.refreshToken)
            .path("/")
            .secure(false)
            .httpOnly(true)
            .sameSite("Lax")
            .maxAge((7 * 24 * 60 * 60).toLong())
            .build()

        response.setHeader("Set-Cookie", responseCookie.toString())
        return ResponseEntity.ok(ApiResponse.success("로그인 성공", tokenResponse.accessToken))
    }

    @PostMapping("/logout")
    fun logout(@AuthenticationPrincipal user: SecurityUser): ResponseEntity<ApiResponse<String>> {
        authService.logout(user.userId)
        return ResponseEntity.ok(ApiResponse.success("로그아웃 성공"))
    }

    @PostMapping("/refresh")
    fun refresh(
        @CookieValue(value = "refreshToken", required = false) refreshToken: String,
        response: HttpServletResponse
    ): ApiResponse<String> {
        val tokenResponse: TokenResponse = authService.refreshAccessToken(refreshToken)
        val responseCookie: ResponseCookie = ResponseCookie.from("refreshToken", tokenResponse.refreshToken)
            .path("/")
            .secure(false) // https 환경에서만 쿠키가 발동합니다.
            .sameSite("Lax") // 동일 사이트과 크로스 사이트에 모두 쿠키 전송이 가능합니다
            .httpOnly(true) // 브라우저에서 쿠키에 접근할 수 없도록 제한
            .maxAge((7 * 24 * 60 * 60).toLong())
            .build()

        response.setHeader("Set-Cookie", responseCookie.toString())
        return ApiResponse.success("토큰 갱신 성공", tokenResponse.accessToken)
    }
}
